src/modbus.c

```diff
@@ -1070,11 +1070,13 @@ int modbus_reply(modbus_t *ctx,
         rsp[byte_count_pos] = rsp_length - byte_count_pos - 1;
     } break;
     case MODBUS_FC_READ_EXCEPTION_STATUS:
-        if (ctx->debug) {
-            fprintf(stderr, "FIXME Not implemented\n");
-        }
-        errno = ENOPROTOOPT;
-        return -1;
+        rsp_length = response_exception(ctx,
+                                        &sft,
+                                        MODBUS_EXCEPTION_ILLEGAL_FUNCTION,
+                                        rsp,
+                                        TRUE,
+                                        "FIXME Not implemented: READ EXCEPTION STATUS (0x07)\n",
+                                        function);
```

Comment:
The `function` argument is not needed anymore. We can drop it.
I also recommend to squash all the commits into a single one and then force push to this branch.